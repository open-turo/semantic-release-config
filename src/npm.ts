import config from "./";

/**
 * Semantic release configuration for NPM projects. It extends the default configuration by publishing to NPM
 * and creating a Pull Request with version updates in files like package.json
 */
export = {
  ...config,
  plugins: [
    ...config.plugins,
    [
      "@semantic-release/npm",
      {
        tarballDir: "pack",
      },
    ],
    [
      "@semantic-release/git",
      {
        assets: ["package.json", "package-lock.json", "README.md", "yarn.lock"],
        message:
          "ci(release): ${nextRelease.version} <% nextRelease.channel !== 'next' ? print('[skip ci]') : print('') %>\n\n${nextRelease.notes}",
      },
    ],
  ],
};
