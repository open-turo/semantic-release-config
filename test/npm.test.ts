import template = require("lodash.template");

import npm from "~/npm";

describe("npm", () => {
  test.each(["next", undefined])(
    "semantic-release/@git generates the right commit message for channel %s",
    (channel) => {
      const gitPlugin = npm.plugins.find(
        (plugin) => plugin[0] === "@semantic-release/git"
      );
      expect(gitPlugin).toBeDefined();
      const pluginConfig = ((gitPlugin && gitPlugin[1]) || { message: "" }) as {
        message: string;
      };
      expect(
        template(pluginConfig.message || "")({
          nextRelease: {
            version: "test",
            channel: channel,
            notes: "This is a test release",
          },
        })
      ).toMatchSnapshot();
    }
  );

  test("@semantic-release/exec and semantic-release-fotingo are the last plugins", () => {
    expect(npm.plugins[npm.plugins.length - 2]).toBe(
      "semantic-release-fotingo"
    );
    expect(npm.plugins[npm.plugins.length - 1][0]).toBe(
      "@semantic-release/exec"
    );
  });
});
