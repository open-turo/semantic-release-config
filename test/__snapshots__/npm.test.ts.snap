// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm semantic-release/@git generates the right commit message for channel next 1`] = `
"ci(release): test 

This is a test release"
`;

exports[`npm semantic-release/@git generates the right commit message for channel undefined 1`] = `
"ci(release): test [skip ci]

This is a test release"
`;
